### ISSUE TITLE
Wrong provider config memoization

### ISSUE THREAD

**Issue by dotsent** [lbeder/health-monitor-rails#19]
I configure my redis provider in initializer as shown in documentation
```
HealthMonitor.configure do |config|
  config.redis.configure do |redis_config|
    redis_config.url = 'redis://user:pass@example.redis.com:90210/'
  end
end
```
But i have error:
Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)

I think this error occurs because self.configure method call in Providers::Base#initialize. This call override configuration from initializer to default values. And HealthMonitor.check create new instance each provider on every health check.

**Comment by lbeder** [comment]
Nice catch! I'm baffled how this issue wasn't reproduced yet. I've pushed a fix, will perform some tests and then release as `7.2.0`,

**Comment by dotsent** [comment]
Ok, but how i can configure provider via initializer? 

Uncaught exception: undefined method `configure' for HealthMonitor::Providers::Redis:Class

**Comment by lbeder** [comment]
How are you trying to configure it? The code you've written above (like in the documentation) should work.

**Comment by lbeder** [comment]
Oh, I see what you mean...

**Comment by lbeder** [comment]
Fixed.

**Comment by dotsent** [comment]
:) And nothing has changed.

Base provider still calls self.configure and overwrites configuration
  
            
 
Write

**Comment by lbeder** [comment]
Are you sure you're using the last version? I have addressed this issue and added a failing (and now passing) test exactly for that.

**Comment by dotsent** [comment]
Yes, my app use v7.2.1.
[This row](https://github.com/lbeder/health-monitor-rails/blob/master/lib/health_monitor/providers/base.rb#L13) create new blank configuration object every `HealthMonitor.check`

**Comment by lbeder** [comment]
Right, but this call only needs to be called once. I'll fix it in any case. Thanks for catching this :).

**Comment by lbeder** [comment]
Another attempt to fix it in v7.2.2. Could you give it a shot?

**Comment by dotsent** [comment]
Thanks. It's work.
